src/CodeGen_Hexagon.h

```diff
@@ -12,6 +12,25 @@ namespace Internal {
 
 /** A code generator that emits Hexagon code from a given Halide stmt. */
 class CodeGen_Hexagon : public CodeGen_Posix {
+    // LLVM's HVX vector intrinsics don't include the type of the
+    // operands, they all operate on vectors of 32 bit integers. To make
+    // it easier to generate code, we define wrapper intrinsics with
+    // the correct type (plus the necessary bitcasts).
+    struct HvxIntrinsic {
+        enum {
+            BroadcastScalarsToWords = 1 << 0,  // Some intrinsics need scalar arguments broadcasted up to 32 bits.
+            InterleaveByteAndBroadcastToWord = 1 << 1, // i8 low, i8 high  = i32 high_low_high_low. Always takes the last two args.
```

Comment:
I think there's so much going on here in the new vmpa instruction wrappers, that they should just be defined in hvx_64.ll and hvx_128.ll, instead of with these new flags.